cstore_fdw.c

```diff
@@ -532,6 +531,45 @@ DroppedCStoreFilenameList(DropStmt *dropStatement)
 }
 
 
+/* FindCStoreTables returns list of CStore tables from given table list */
+static List *
+FindCStoreTables(List *tableList)
+{
+	List *cstoreTableList = NIL;
+	ListCell *relationCell = NULL;
+	foreach(relationCell, tableList)
+	{
+		RangeVar *rangeVar = (RangeVar *) lfirst(relationCell);
+
+		Oid relationId = RangeVarGetRelid(rangeVar, AccessShareLock, true);
+		if (CStoreTable(relationId))
+		{
+			cstoreTableList = lappend(cstoreTableList, rangeVar);
+		}
+	}
+
+	return cstoreTableList;
+}
+
+/* TruncateCStoreTable truncates given cstore tables */
+static void
+TruncateCStoreTables(List *cstoreTableList)
+{
+	ListCell *relationCell = NULL;
+	foreach(relationCell, cstoreTableList)
+	{
+		RangeVar *rangeVar = (RangeVar *) lfirst(relationCell);
```

Comment:
In postgres documentation (and source) it says that `truncate` [acquires](http://www.postgresql.org/docs/9.5/static/sql-truncate.html) `ACCESS EXCLUSIVE lock `. I think we should definitely acquire AccessExcluiveLock as we acquire `ExclusiveLock` [here](https://github.com/citusdata/cstore_fdw/blob/feature-9.5_compatibility/cstore_fdw.c#L199). 
Or, maybe we can use `RangeVarGetRelid ` with `AccessExclusiveLock`, which seems equivalent, but I am not sure.

(Note: Postgres acquires the lock as [here](https://github.com/postgres/postgres/blob/ee943004466418595363d567f18c053bae407792/src/backend/commands/tablecmds.c#L1022))